File: spark/src/test/scala/geotrellis/spark/io/file/FileSpaceTimeSpec.scala
/*
 * Copyright 2016 Azavea
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package geotrellis.spark.io.file

import geotrellis.tiling._
import geotrellis.raster.Tile
import geotrellis.layers.TileLayerMetadata
import geotrellis.layers.file._
import geotrellis.layers.index._
import geotrellis.spark._
import geotrellis.spark.io._
import geotrellis.spark.io.file._
import geotrellis.spark.testkit._
import geotrellis.spark.testkit.io._
import geotrellis.spark.testkit.testfiles.TestFiles


class FileSpaceTimeSpec
    extends PersistenceSpec[SpaceTimeKey, Tile, TileLayerMetadata[SpaceTimeKey]]
    with SpaceTimeKeyIndexMethods
    with TestEnvironment
    with TestFiles
    with CoordinateSpaceTimeSpec
    with LayerUpdateSpaceTimeTileSpec {
  lazy val reader = FileLayerReader(outputLocalPath)
  lazy val creader = FileCollectionLayerReader(outputLocalPath)
  lazy val writer = FileLayerWriter(outputLocalPath)
  lazy val deleter = FileLayerDeleter(outputLocalPath)
  lazy val copier = FileLayerCopier(outputLocalPath)
  lazy val mover  = FileLayerMover(outputLocalPath)
  lazy val reindexer = FileLayerReindexer(outputLocalPath)
  lazy val tiles = FileValueReader(outputLocalPath)
  lazy val sample =  CoordinateSpaceTime
}

File: geotrellis-python/tests/geotrellis/spark/io/file/FileSpaceTimeSpec.py
from spec import Spec
from tests.geotrellis.spark.io.PersistenceSpec import PersistenceSpec
from tests.geotrellis.spark.io.SpaceTimeKeyIndexMethods import _SpaceTimeKeyIndexMethods
from tests.geotrellis.spark.TestEnvironment import _TestEnvironment
from tests.geotrellis.spark.testfiles.TestFiles import _TestFiles
from geotrellis.spark.SpatialKey import SpatialKey
from geotrellis.raster.Tile import Tile
from geotrellis.spark.TileLayerMetadata import TileLayerMetadata
from geotrellis.spark.io.file.FileLayerReader import FileLayerReader
from geotrellis.spark.io.file.FileLayerWriter import FileLayerWriter
from geotrellis.spark.LayerId import LayerId
from geotrellis.spark.io.index.ZCurveKeyIndexMethod import ZCurveKeyIndexMethod

class FileSpaceTimeSpec(
        Spec,
        PersistenceSpec[SpatialKey, Tile, TileLayerMetadata],
        _SpaceTimeKeyIndexMethods,
        _TestEnvironment,
        _TestFiles):

    @property
    def reader(self):
        return FileLayerReader(self.outputLocalPath)

    @property
    def writer(self):
        return FileLayerWriter(self.outputLocalPath)
